packages/python-google-compute-engine/google_compute_engine/metadata_scripts/script_retriever.py

```diff
@@ -81,8 +95,8 @@ def _DownloadAuthUrl(self, url, dest_dir):
       request = urlrequest.Request(url)
       request.add_unredirected_header('Metadata-Flavor', 'Google')
       request.add_unredirected_header('Authorization', self.token)
-      content = urlrequest.urlopen(request).read().decode('utf-8')
-    except (httpclient.HTTPException, socket.error, urlerror.URLError) as e:
+      content = UrlOpenWithRetry(request).read().decode('utf-8')
+    except Exception as e:
```

Comment:
the concrete exception type is extracted to the common "RetryOnUnavailable" method in network_utils

Reply:
I changed the behavior here intentionally: I catch "httpclient.HTTPException, socket.error, urlerror.URLError" in the retry logic to consider them as "retriable network issue".
For any other exceptions: since an exception is raised, the downloading very likely is failed, so I treat them as "Could not download".

Is this a correct change? If not, ,may I know why we should consider it as a successful download when getting an exception?

Reply:
That's fine, thanks!